src/Exception/BadResponseException.php

```diff
@@ -33,6 +33,9 @@ public function hasResponse(): bool
      */
     public function getResponse(): ResponseInterface
     {
-        return parent::getResponse();
+        /** @var ResponseInterface $response */
+        $response = parent::getResponse();
+
+        return $response;
```

Comment:
I might say something that might not be accurate in the slightest.. and I am not that familiar with the deep internals of PHP... personally if this has even the slightest performance impact I prefer having a quick return and a baseline error rather than having a clean baseline.

Reply:
You can just write:

```
/** @var ResponseInterface */
return parent::getResponse();
```

and phpstan will work out what you meant. No need to assign the variable.

Reply:
Thank you for mentioning this @GrahamCampbell 👏 Agree with not having to define a variable.

I was still living under assumption that PHPStan did not properly support this yet, but they might have changed / improved this in the meanwhile.

Fixed in 734e50e667101f58d560ce93207814a7cb09fe8a